trillian/integration/ct_hammer/main.go

```diff
@@ -247,3 +333,33 @@ func main() {
 	}
 	glog.Info("  no errors; done")
 }
+
+// notAfterForLog returns a notAfter time to be used for certs submitted
+// to the given log instance, allowing for any temporal shard configuration.
+func notAfterForLog(c *configpb.LogConfig) (time.Time, error) {
+	if c.NotAfterStart == nil && c.NotAfterLimit == nil {
+		return time.Now().Add(24 * time.Hour), nil
+	}
+
+	if c.NotAfterStart != nil {
+		start, err := ptypes.Timestamp(c.NotAfterStart)
+		if err != nil {
+			return time.Time{}, fmt.Errorf("failed to parse NotAfterStart: %v", err)
+		}
+		if c.NotAfterLimit == nil {
+			return start.Add(24 * time.Hour), nil
```

Comment:
Is 24 hours arbitrary? Should it be variable?

Reply:
It is arbitrary (and pre-existing, but that's hard to see because things have moved around) but I don't think it needs to be variable (it would only matter if we ran a >24h test on a temporal shard with a semi-infinite [T, ∞) `NotAfter` interface, which has never come up so far...)